### PR TITLE
Fix for Group Policy enforced include/exclude lists not being imported corrected

### PR DESCRIPTION
**Pull request by nickjisc**
# Proposed Changes

Fix for Group Policy enforced include/exclude lists not being imported corrected.

Previously the GPO allow/block list wasn't getting imported correctly, (empty array), this fixes that by using native powershell registry calls

### CODE REVIEW

File: Sources/WAU/Winget-AutoUpdate/functions/Get-ExcludedApps.ps1 [hunk 1]
```diff
@@ -10,16 +10,14 @@ function Get-ExcludedApps {
 
             $ValueNames = (Get-Item -Path "HKLM:\SOFTWARE\Policies\Romanitho\Winget-AutoUpdate\BlackList").Property
 
+            $AppIDs = @()
+
             foreach ($ValueName in $ValueNames) {
-                $AppIDs = [Microsoft.Win32.Registry]::GetValue($Key, $ValueName, $false)
-                [PSCustomObject]@{
-                    Value = $ValueName
-                    Data  = $AppIDs.Trim()
```
Comment (Romanitho) [inline Sources/WAU/Winget-AutoUpdate/functions/Get-ExcludedApps.ps1:17]: Shouldn't we keep the Trim control?

File: Sources/WAU/Winget-AutoUpdate/functions/Get-IncludedApps.ps1 [hunk 2]
```diff
@@ -10,12 +10,10 @@ function Get-IncludedApps {
 
             $ValueNames = (Get-Item -Path "HKLM:\SOFTWARE\Policies\Romanitho\Winget-AutoUpdate\WhiteList").Property
 
+            $AppIDs = @()
+
             foreach ($ValueName in $ValueNames) {
-                $AppIDs = [Microsoft.Win32.Registry]::GetValue($Key, $ValueName, $false)
-                [PSCustomObject]@{
-                    Value = $ValueName
-                    Data  = $AppIDs.Trim()
```
Comment (Romanitho) [inline Sources/WAU/Winget-AutoUpdate/functions/Get-IncludedApps.ps1:17]: Shouldn't we keep the Trim control?

Comment (nickjisc) [reply Sources/WAU/Winget-AutoUpdate/functions/Get-IncludedApps.ps1:17]: Add that back in and updated the PR

File: Sources/WAU/Winget-AutoUpdate/Winget-Upgrade.ps1 [hunk 3]
```diff
@@ -245,7 +245,8 @@ if (Test-Network) {
                     New-Item "$WorkingDir\logs\error.txt" -Value "Whitelist doesn't exist in GPO" -Force
                     Exit 1
                 }
-                $toUpdate = $toUpdate.Data
+                $toUpdate = $toUpdate
```
Comment (Romanitho) [inline Sources/WAU/Winget-AutoUpdate/Winget-Upgrade.ps1:248]: This could be removed ?

File: Sources/WAU/Winget-AutoUpdate/Winget-Upgrade.ps1 [hunk 4]
```diff
@@ -254,7 +255,8 @@ if (Test-Network) {
                     New-Item "$WorkingDir\logs\error.txt" -Value "Blacklist doesn't exist in GPO" -Force
                     Exit 1
                 }
-                $toSkip = $toSkip.Data
+                $toSkip = $toSkip
```
Comment (Romanitho) [inline Sources/WAU/Winget-AutoUpdate/Winget-Upgrade.ps1:258]: Same ?

**Review by Romanitho — COMMENTED** [review]
Another thing 😇

File: Sources/WAU/Winget-AutoUpdate/Winget-Upgrade.ps1 (hunk 4 above)
Comment (nickjisc) [reply Sources/WAU/Winget-AutoUpdate/Winget-Upgrade.ps1]: yep that'd be the same, I've removed that whole line now

**Comment by AndrewDemski-ad-gmail-com** [comment]
I went through different path..
Maybe it s worth a look
https://github.com/Romanitho/Winget-AutoUpdate/pull/586

**Comment by nickjisc** [comment]
Added the uri fetching.  It does seem odd you are creating custom objects (key value pair) then ignoring those in the main code base, question if that is needed or you just need an array of the items, which is what I'm doing.

I'm mainly focused on the excluded apps section, as the original code wasn't fully working.  New code does seem happy though it can always be slightly optimized

**Comment by AndrewDemski-ad-gmail-com** [comment]
II just expanded the function. How it is handled RN in the calling scriptblock was not adjusted at 11PM. I am aware of ongoing discussion about that $appIds.Data thread.

BR
AD
________________________________

**Pull request merged**